### PR TITLE
Add semantic color keywords and improve formatting of color examples

### LINKED ISSUE #1522: semantic colorschemes & color keywords
thoughts on adding a few new colorschemes?

semantic | color
---|---
error |  red
notice | orange
success | brightgreen
info | blue
default | lightgrey

currently, setting the color is completely non-semaantic

Also: is there a reason [color keywords](https://www.w3.org/TR/SVG/types.html#ColorKeywords) don't work?

[doc/gh-badges.md](https://github.com/badges/shields/blob/master/doc/gh-badges.md#defaults) ("CSS/SVG Color") links to  http://www.w3.org/TR/SVG/types.html#DataTypeColor
which in turn references the color keywords doc:  https://www.w3.org/TR/SVG/types.html#ColorKeywords

### PR DESCRIPTION
**Pull request by paulmelnikow**
This is based on the list I proposed at #1522 (comment). As I started documenting `default` I realized it didn't feel quite right. It's not semantic in relation to the content the way the others are, and it's also not the default left color. I changed it to `disabled` which isn't perfect, but seems better. I'm open to other suggestions.

I updated `gh-badges/readme.md` but the colors won't render correctly until this is deployed

Close #1522

### CODE REVIEW

File: core/badge-urls/make-badge-url.js [hunk 1]
```diff
@@ -61,8 +61,8 @@ function staticBadgeUrl({
   style,
   format = 'svg',
 }) {
-  if (!label || !message) {
-    throw Error('label and message are required')
+  if (!message) {
```
Comment (calebcartwright) [inline core/badge-urls/make-badge-url.js]: What are the implications of `label` no longer being required?

Comment (paulmelnikow) [reply core/badge-urls/make-badge-url.js]: It lets you use this function to make a badge without a label, nothing beyond that.

Comment (calebcartwright) [reply core/badge-urls/make-badge-url.js]: 👍

**Comment by calebcartwright** [comment]
I'm still working my way through `color.js`, but on the alias for `default` it looks like you went with `inactive` instead right? `inactive` or `disabled` both make sense to me. The only other word I mentally associate with that color is `inaccessible` but not sure that makes sense in this context

File: gh-badges/lib/color.js [hunk 2]
```diff
@@ -29,6 +42,8 @@ function normalizeColor(color) {
     return undefined
   } else if (color in namedColors) {
     return color
+  } else if (color in aliases) {
+    return aliases[color]
```
Comment (calebcartwright) [inline gh-badges/lib/color.js]: Could we use `resolvedAliases` here instead? It's been a long day and my brain is struggling sorry 😄

Comment (paulmelnikow) [reply gh-badges/lib/color.js]: The idea of this function is to take the input color and return something that is normalized, yet still human readable and pleasant to test. For named colors it passes through their names; they don't translate to their hex equivalents. So, the appropriate response for the aliases is the sort of normalized / canonical color name (e.g. `red` -> `red`; `critical` -> `red`).

**Comment by paulmelnikow** [comment]
> I'm still working my way through `color.js`, but on the alias for `default` it looks like you went with `inactive` instead right? `inactive` or `disabled` both make sense to me. The only other word I mentally associate with that color is `inaccessible` but not sure that makes sense in this context

Aha yes, `inactive`. I like that better than `disabled` which is what I had mistakenly written in the top comment. If it were an action button it would be clearer, but as an informational button that's not displaying any information… what do you call that? Inactive is the best name I've thought of for that situation.

While `unknown` could work semantically, as a color specification, it seems really confusing. We do use this color when information is inaccessible, though `inaccessible` again feels like an odd way to specify it.

We could be really explicit and call it something like `information-missing` or `no-content`.

File: gh-badges/lib/color.js (hunk 2 above)
Comment (calebcartwright) [reply gh-badges/lib/color.js]: Ah okay, thanks for the explanation. I was looking at it from the perspective of being called from within `toSvgColor` but totally missed that these can (and I believe are) called independently

**Comment by paulmelnikow** [comment]
@RedSparr0w @chris48s Any thoughts / objections?

**Comment by paulmelnikow** [comment]
If there are no objections I'll merge this at the end of the weekend.

**Comment by paulmelnikow** [comment]
@calebcartwright want to merge this?

**Comment by chris48s** [comment]
Didn't get a chance to look at this one before it was merged. Broadly I'm on board with it. I find "inactive" a confusing label, but I'm not sure I have a better suggestion :man_shrugging:

**Comment by paulmelnikow** [comment]
Hasn't deployed yet so it's still a good time to tweak it. Among the alternatives, `no-content` is my favorite, though not sure it's better.

**Pull request merged**